core/services/llo/onchain_channel_definition_cache.go

```diff
@@ -0,0 +1,260 @@
+	filterName := logpoller.FilterName("OCR3 LLO ChannelDefinitionCachePoller", addr.String())
+	return &channelDefinitionCache{
+		services.StateMachine{},
+		orm,
+		filterName,
+		lp,
+		0,
+		addr,
+		lggr.Named("ChannelDefinitionCache").With("addr", addr, "fromBlock", fromBlock),
+		sync.RWMutex{},
+		nil,
+		fromBlock,
+		sync.WaitGroup{},
+		make(chan struct{}),
+	}
+}
+
+func (c *channelDefinitionCache) Start(ctx context.Context) error {
+	// Initial load from DB, then async poll from chain thereafter
+	return c.StartOnce("ChannelDefinitionCache", func() (err error) {
+		err = c.lp.RegisterFilter(logpoller.Filter{Name: c.filterName, EventSigs: allTopics, Addresses: []common.Address{c.addr}}, pg.WithParentCtx(ctx))
+		if err != nil {
+			return err
+		}
+		if definitions, definitionsBlockNum, err := c.orm.LoadChannelDefinitions(ctx, c.addr); err != nil {
+			return err
+		} else if definitions != nil {
+			c.definitions = definitions
+			c.definitionsBlockNum = definitionsBlockNum
+		} else {
+			// ensure non-nil map ready for assignment later
+			c.definitions = make(commontypes.ChannelDefinitions)
+			// leave c.definitionsBlockNum as provided fromBlock argument
+		}
+		c.wg.Add(1)
+		go c.poll()
+		return nil
+	})
+}
+
+// TODO: make this configurable?
+const pollInterval = 1 * time.Second
+
+func (c *channelDefinitionCache) poll() {
+	defer c.wg.Done()
+
+	pollT := time.NewTicker(utils.WithJitter(pollInterval))
+
+	for {
+		select {
+		case <-c.chStop:
+			return
+		case <-pollT.C:
+			if n, err := c.fetchFromChain(); err != nil {
+				// TODO: retry with backoff?
+				// https://smartcontract-it.atlassian.net/browse/MERC-3653
+				c.lggr.Errorw("Failed to fetch channel definitions from chain", "err", err)
+				continue
+			} else {
+				if n > 0 {
+					c.lggr.Infow("Updated channel definitions", "nLogs", n, "definitionsBlockNum", c.definitionsBlockNum)
+				} else {
+					c.lggr.Debugw("No new channel definitions", "nLogs", 0, "definitionsBlockNum", c.definitionsBlockNum)
+				}
+			}
+		}
+	}
+}
+
+func (c *channelDefinitionCache) fetchFromChain() (nLogs int, err error) {
+	// TODO: Pass context
+	// https://smartcontract-it.atlassian.net/browse/MERC-3653
+	latest, err := c.lp.LatestBlock()
+	if errors.Is(err, sql.ErrNoRows) {
+		return 0, nil
+	} else if err != nil {
+		return 0, err
+	}
+	toBlock := latest.BlockNumber
+
+	fromBlock := c.definitionsBlockNum
+
+	if toBlock <= fromBlock {
+		return 0, nil
+	}
+
+	ctx, cancel := services.StopChan(c.chStop).NewCtx()
+	defer cancel()
+	// NOTE: We assume that log poller returns logs in ascending order chronologically
+	logs, err := c.lp.LogsWithSigs(fromBlock, toBlock, allTopics, c.addr, pg.WithParentCtx(ctx))
+	if err != nil {
+		// TODO: retry?
+		// https://smartcontract-it.atlassian.net/browse/MERC-3653
+		return 0, err
+	}
+	for _, log := range logs {
+		if err = c.applyLog(log); err != nil {
+			return 0, err
```

Comment:
do you want to accumulate all the errors or quit on the first one?

Reply:
Any error here means our channel definition cache is busted and can't be caught up to what it should be, it feels safer to bomb out immediately.

This is kind of like applying git commits, if one fails it doesnt make much sense to keep applying commits on top of whatever broke.